### PR TITLE
Update extension tester version after API changes in 4.0.1

### PR DESCRIPTION
**Pull request by odockal**
- refactor UI test suites
   - adjust to the latest changes in extension

Signed-off-by: Ondrej Dockal <odockal@redhat.com>

### CODE REVIEW

File: test/ui-test/custom-settings.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "files.simpleDialog.enable": false,
+    "window.dialogStyle": "custom"
+}
```
Comment (joshuawilson) [inline test/ui-test/custom-settings.json:4]: this file needs a new line at the end

File: test/ui-test/extensionInstallationUITest.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      const sideBar = await viewBar.openView();
+      const section = (await sideBar.getContent().getSection('Installed')) as ExtensionsViewSection;
+      const item = await driver.wait(async () => section.findItem(`@installed ${KNativeConstants.KNATIVE_EXTENSION_NAME}`), 3000);
+      expect(item).to.be.an.instanceOf(ExtensionsViewItem);
+      expect(await item.getTitle()).to.equal(KNativeConstants.KNATIVE_EXTENSION_NAME);
+    });
+
+    it('Activity Bar should be available', async function context() {
+      this.timeout(5000);
+      try {
```
Comment (joshuawilson) [inline test/ui-test/extensionInstallationUITest.ts:30]: Where is the `expect` that the test must pass?

File: test/ui-test/extensionUITest.ts [hunk 3]
```diff
@@ -130,6 +90,20 @@ export function extensionsUITest(): void {
       });
     });
 
+    if (!clusterIsAvailable) {
+      it('should notify user that he must log into a cluster', async function context() {
```
Comment (joshuawilson) [inline test/ui-test/extensionUITest.ts:94]: If this is setup, then it does not need to be in an `it`. If it is a test, where is the `expect`?

File: test/ui-test/extensionInstallationUITest.ts (hunk 2 above)
Comment (odockal) [reply test/ui-test/extensionInstallationUITest.ts:30]: Assert here is implicated by not failing the test in catch block.

File: test/ui-test/extensionUITest.ts (hunk 3 above)
Comment (odockal) [reply test/ui-test/extensionUITest.ts:94]: not a setup, it is a check for dialog that talks to user. but I will add expect, here it makes sense.

File: test/ui-test/extensionInstallationUITest.ts (hunk 2 above)
Comment (joshuawilson) [reply test/ui-test/extensionInstallationUITest.ts:30]: So the `fail` will register with the test framework?

File: test/ui-test/extensionUITest.ts (hunk 3 above)
Comment (joshuawilson) [reply test/ui-test/extensionUITest.ts:94]: ok, thanks

File: test/ui-test/extensionInstallationUITest.ts (hunk 2 above)
Comment (odockal) [reply test/ui-test/extensionInstallationUITest.ts:30]: Hm, it should be this way. Always was, but I did not check :D

Comment (joshuawilson) [reply test/ui-test/extensionInstallationUITest.ts:30]: do we need a test for the test? ;-)

Comment (odockal) [reply test/ui-test/extensionInstallationUITest.ts:30]: I really hope not! :D

**Comment by odockal** [comment]
@joshuawilson Now comes long and boring debugging of tests failing on CI...

**Comment by odockal** [comment]
@joshuawilson I have one test failing with my tailored failure message: `1) should notify user that he must log into a cluster`. It expects when knative is activated (knative tab opened) is shows a dialog (custom / native) that user is not logged in. It was introduced here [1]. It this still working? What are circumstances to get that dialog? Or is not out of game now?

[1]: https://github.com/talamer/vscode-knative/commit/9b6bc9fa86bf815540bf536178ef21d241427176#diff-674ffa732ca648a26dbeb82cab704330260cec2574a9f65991157b38e9e0c709R96

**Comment by odockal** [comment]
@joshuawilson Got a chance to take a look at the failing test case? feature / bug paradigm :)

**Comment by joshuawilson** [comment]
If I have a kubeconfig for cluster that is not running, then I get the message `The cluster is not up. Please log into a running cluster.`

Currently it can only login via kubeconfig, so it must be one that matches a running cluster.

**Comment by odockal** [comment]
@sudhirverma If you could plese do a review, I can merge it then, so we got green builds.

**Comment by sudhirverma** [comment]
> @sudhirverma If you could plese do a review, I can merge it then, so we got green builds.

okay, I am reviewing your PR :)

**Comment by sudhirverma** [comment]
I got two errors when I was trying to run the ui-test.

Screenshot 2021-09-17 at 8 26 16 PM

**Comment by odockal** [comment]
It happens when you have downloaded kn cli already. It should be enough to delete ~/.vs-kn folder. On fresh env. on CI this wont happen. that is why tests pass there.

**Review by sudhirverma — APPROVED** [review]

**Pull request merged**